autogpt/config/ai_config.py

```diff
@@ -79,14 +77,15 @@ def save(self, config_file: str = SAVE_FILE) -> None:
         Returns:
             None
         """
+        config_file = Path(config_file)
 
         config = {
             "ai_name": self.ai_name,
             "ai_role": self.ai_role,
             "ai_goals": self.ai_goals,
         }
-        with open(config_file, "w", encoding="utf-8") as file:
-            yaml.dump(config, file, allow_unicode=True)
+        config_file.parent.mkdir(parents=True, exist_ok=True)
```

Comment:
What's this for?

Reply:
From the scope of the function, don't assume that ai_settings is going to be in the same location all the time then you will see somebody could pass in "a/b/c/d/some_settings.yaml" and also that directory path might not exist and will error out if it doesn't exist.  This is a recursive mkdir that does it without questions.